File: app/src/main/java/net/emojiparty/android/jishotomo/ui/viewmodels/PagedEntriesViewModelFactory.java
package net.emojiparty.android.jishotomo.ui.viewmodels;

import android.app.Application;
import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.support.annotation.NonNull;
import net.emojiparty.android.jishotomo.data.room.EntryDao;

public class PagedEntriesViewModelFactory extends ViewModelProvider.NewInstanceFactory {
  private Application application;
  @NonNull private EntryDao entryDao;

  public PagedEntriesViewModelFactory(Application application, EntryDao entryDao) {
    this.application = application;
    this.entryDao = entryDao;
  }

  @NonNull @Override public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
    return (T) new PagedEntriesViewModel(application, entryDao);
  }
}

File: app/src/main/java/net/emojiparty/android/jishotomo/ui/viewmodels/PagedEntriesViewModelFactory.kt
package net.emojiparty.android.jishotomo.ui.viewmodels

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import net.emojiparty.android.jishotomo.ui.presentation.ResourceFetcher

class PagedEntriesViewModelFactory(
  private val resourceFetcher: ResourceFetcher
) : ViewModelProvider.Factory {
  @Suppress("UNCHECKED_CAST")
  override fun <T : ViewModel> create(modelClass: Class<T>): T {
    return PagedEntriesViewModel(resourceFetcher) as T
  }
}
